common/rulesys.cpp

```diff
@@ -546,139 +521,83 @@ bool RuleManager::UpdateOrphanedRules(Database *db, bool quiet_update)
 
 bool RuleManager::RestoreRuleNotes(Database *db)
 {
-	std::string query("SELECT `ruleset_id`, `rule_name`, `notes` FROM `rule_values`");
-
-	auto results = db->QueryDatabase(query);
-	if (!results.Success()) {
+	const auto& l = RuleValuesRepository::All(*db);
+	if (l.empty()) {
 		return false;
 	}
 
 	int update_count = 0;
-	for (auto row = results.begin(); row != results.end(); ++row) {
-
-		auto rule = [](const char *rule_name) {
+	for (const auto& e : l) {
 
+		auto rule = [](std::string rule_name) {
 			for (auto rule_iter : s_RuleInfo) {
-				if (strcasecmp(rule_iter.name, rule_name) == 0) {
+				if (Strings::EqualFold(rule_iter.name, rule_name)) {
 					return rule_iter;
 				}
 			}
 
-			return s_RuleInfo[_IntRuleCount+_RealRuleCount+_BoolRuleCount];
-		}(row[1]);
+			return s_RuleInfo[_IntRuleCount + _RealRuleCount + _BoolRuleCount];
+		}(e.rule_name);
 
-		if (strcasecmp(rule.name, row[1]) != 0) {
+		if (Strings::Contains(rule.name, e.rule_name)) {
```

Comment:
This was previously an exact match but is partial now, guessing this is unintentional